### DISCUSSION TITLE
Bootstrap overwrites HTML href scrolling

### DISCUSSION THREAD

**Discussion by AdamRatcliffeDeetu** [twbs/bootstrap#39389] [Q&A]
Posting this here first because I'm not sure if it's a genuine bug yet or simply an issue with implementation.
I'm working on a project that uses the HTML function to scroll to a specific part of a page if the href attribute contains the ID of an element. We'd recently updated from v1.1 to v5.3 because of the new Accordian functionality provided. After updating existing projects to the most recent version we encountered an issue that causes this scrolling to not work on some elements that function as the chapters of a story.

We also discovered that there is no consistency between the projects we update and the chapters that become affected.

I had implemented a fix using JQuery with the following:
```js
var offsets = [];
var mobileWidth = 440;

$("a[href^='#']").on('click', function () {
  if(window.innerWidth < mobileWidth) return;
  $('#features').animate({
    scrollTop: offsets[$(this).attr('href').substring(1)]
  }, 500);
});

$("section").each(function () {
  if(window.innerWidth < mobileWidth) return;
  offsets[$(this).attr('id')] = $(this).offset().top;
});
```
This worked fine for the project I was working on as a test project however, when the fix was implemented on other projects that were suffering from the issue outlined above, the issue became that the scroll would not scroll down all the way to the actual top of the element and instead was off by quite a sizeable margin.

If there's any possible fixes for this issue I'd greatly appreciate it. I have a feeling this was something we should have encountered earlier if we'd kept our Bootstrap version up to date.

**Comment by julien-deramond** [comment]
Hey @AdamRatcliffeDeetu 
Would you be able to create a reduced test case on [CodePen](https://codepen.io/) or [StackBlitz](https://stackblitz.com/)? It would be helpful to start with something and to be sure to understand precisely the issue you encounter :)
